src/platform/ESP32/OpenthreadLauncher.cpp

```diff
@@ -20,18 +20,92 @@
 #include "esp_netif.h"
 #include "esp_netif_types.h"
 #include "esp_openthread.h"
+#include "esp_openthread_cli.h"
 #include "esp_openthread_lock.h"
 #include "esp_openthread_netif_glue.h"
 #include "esp_openthread_types.h"
 #include "esp_vfs_eventfd.h"
 #include "freertos/FreeRTOS.h"
 #include "freertos/task.h"
+#include "openthread/cli.h"
 #include "openthread/instance.h"
 #include "openthread/logging.h"
 #include "openthread/tasklet.h"
+#include <lib/core/CHIPError.h>
+#include <memory>
 
 static esp_netif_t * openthread_netif                       = NULL;
 static esp_openthread_platform_config_t * s_platform_config = NULL;
+static TaskHandle_t cli_transmit_task                       = NULL;
+static QueueHandle_t cli_transmit_task_queue                = NULL;
+
+CHIP_ERROR cli_transmit_task_post(std::unique_ptr<char[]> && cli_str)
+{
+    char * cmd = cli_str.get();
+    if (!cli_transmit_task_queue || xQueueSend(cli_transmit_task_queue, &cmd, portMAX_DELAY) != pdTRUE)
+    {
+        return CHIP_ERROR_INTERNAL;
+    }
+    cli_str.release();
+    return CHIP_NO_ERROR;
+}
+
+static int cli_output_callback(void * context, const char * format, va_list args)
+{
+    int ret = 0;
+    char prompt_check[3];
+    vsnprintf(prompt_check, sizeof(prompt_check), format, args);
+    if (!strncmp(prompt_check, "> ", sizeof(prompt_check)) && cli_transmit_task)
+    {
+        xTaskNotifyGive(cli_transmit_task);
+    }
+    else
+    {
+        ret = vprintf(format, args);
+    }
+    return ret;
+}
+
+static void esp_openthread_matter_cli_init(void)
+{
+    otCliInit(esp_openthread_get_instance(), cli_output_callback, NULL);
+}
+
+static void cli_transmit_worker(void * context)
+{
+    cli_transmit_task_queue = xQueueCreate(8, sizeof(char *));
+    if (!cli_transmit_task_queue)
+    {
+        vTaskDelete(NULL);
+        return;
+    }
+
+    while (true)
+    {
+        char * cmd = NULL;
+        if (xQueueReceive(cli_transmit_task_queue, &cmd, portMAX_DELAY) == pdTRUE)
+        {
+            if (cmd)
+            {
+                std::unique_ptr<char[]> cmd_ptr(cmd);
+                esp_openthread_cli_input(cmd_ptr.get());
+            }
+            else
+            {
+                continue;
+            }
+            xTaskNotifyWait(0, 0, NULL, portMAX_DELAY);
+        }
+    }
+    vQueueDelete(cli_transmit_task_queue);
+    vTaskDelete(NULL);
+}
+
+static esp_err_t cli_command_transmit_task(void)
+{
+    xTaskCreate(cli_transmit_worker, "ocli_trans", 3072, xTaskGetCurrentTaskHandle(), 5, &cli_transmit_task);
```

Comment:
please create a constant for `5` and `3072` to give them human readable names like stack size and task priority (and maybe explaining how their sizes/values were picked)